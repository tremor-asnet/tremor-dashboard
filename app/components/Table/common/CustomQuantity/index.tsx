import { OrderProduct } from "@/types";

interface QuantityProps {
  products: OrderProduct[];
}

export const CustomQuantity = ({ products }: QuantityProps) => (
  <>
    {products?.map(product => (
      <p
        key={product.id}
        className="py-0.5 text-xs dark:text-white font-semibold leading-[15px] tracking-[0.4px] max-w-[100px] lg:max-w-[200px] xl:max-w-[300px] 2xl:max-w-[400px] min-w-[100px] order-product">
        {product.count}
      </p>
    ))}
  </>
);
