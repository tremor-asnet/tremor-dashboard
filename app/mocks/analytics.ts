export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: "400",
    value: "$143,960",
    bounce: "56.83%",
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    totalAmount: 55,
    duration: "than last week",
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    totalAmount: 3,
    duration: "than last month",
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    totalAmount: 1,
    duration: "than yesterday",
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    totalAmount: 0,
    duration: "Just updated",
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};
