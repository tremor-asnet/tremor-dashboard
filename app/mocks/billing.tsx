// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
