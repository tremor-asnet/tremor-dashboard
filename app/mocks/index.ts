export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
