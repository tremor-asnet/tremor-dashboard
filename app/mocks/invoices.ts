import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
