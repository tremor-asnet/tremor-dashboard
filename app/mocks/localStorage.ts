// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
