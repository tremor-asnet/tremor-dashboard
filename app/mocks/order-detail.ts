export const ORDER_DETAIL_DATA = [
  {
    id: 10421,
    order_code: "KF332",
    order_deliver_price: 14,
    order_tax: "1.95",
    created_at: "1 Nov, 10:20 AM",
    products: [
      {
        id: 1,
        name: "Gold Glasses",
        count: 1,
        price: 40,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/product-12.0b55635d.jpg",
      },
    ],
    track_order_info: {
      generate_order_id: "#1832412",
      generate_order_at: "1 Nov, 10:20 AM",
      transmited_to_courier_at: "1 Nov, 11:20 AM",
      delivered_at: "1 Nov, 10:20 PM",
    },
    billing_info: {
      last4_bank: "1234",
      owner_name: "Oliver Liam",
      company_name: "Viking Burrito",
      email: "oliver@burrito.com",
      vat: "FRB1235476",
    },
  },
];
