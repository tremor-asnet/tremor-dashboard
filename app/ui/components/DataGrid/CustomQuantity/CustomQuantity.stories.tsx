import type { Meta, StoryObj } from "@storybook/react";

// Components
import CustomQuantity from "./index";

// Mocks
import { MOCK_ORDERS } from "@/mocks";

const meta = {
  title: "Components/Tables/CustomQuantity",
  component: CustomQuantity,
  tags: ["autodocs"],
  argTypes: {
    products: {
      description:
        "This is an array include products with a product have type OrderProduct(id: number; name: string; count?: number; price?: number; url?: string)",
    },
  },
} as Meta<typeof CustomQuantity>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    products: MOCK_ORDERS[0].products,
  },
};
