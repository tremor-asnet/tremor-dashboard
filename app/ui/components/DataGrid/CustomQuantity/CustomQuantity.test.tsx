import { render } from "@testing-library/react";

// Components
import CustomQuantity from ".";

// Mocks
import { MOCK_ORDERS } from "@/mocks";

const CustomQuantityComponent = () =>
  render(<CustomQuantity products={MOCK_ORDERS[0].products} />);

describe("CustomQuantity component", () => {
  it("should render snapshot correctly", () => {
    const { container } = CustomQuantityComponent();

    expect(container).toMatchSnapshot();
  });

  it("should renders list with correct products", () => {
    const { getByText } = CustomQuantityComponent();

    MOCK_ORDERS[0].products.forEach(({ count }) => {
      const productCountText = getByText(count.toString());
      expect(productCountText).toBeTruthy();
    });
  });
});
