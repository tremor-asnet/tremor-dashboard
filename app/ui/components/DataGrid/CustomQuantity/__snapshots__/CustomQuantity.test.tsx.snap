// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomQuantity component should render snapshot correctly 1`] = `
<div>
  <p
    class="py-0.5 text-xs dark:text-white font-semibold leading-[15px] tracking-[0.4px] max-w-[50px] lg:max-w-[150px] xl:max-w-[250px] 2xl:max-w-[350px] min-w-[50px] order-product py-5 first:pt-0 last:pb-0 border-0 dark:border-grayish border-b border-gray-100 last:border-0"
  >
    2
  </p>
</div>
`;
